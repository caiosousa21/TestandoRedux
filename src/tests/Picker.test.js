import React from 'react';
import Picker from '../components/Picker';
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() })

describe('index', () => {
    it('Renderizando index', () => {
        const wrapper = shallow(<Picker value='' options={[]} />);
        expect(wrapper.exists()).toBe(true);
    })
    it('Testando interface com snapshots', () => {
        const tree = renderer.create(<Picker value='' options={[]} />).toJSON();
        expect(tree).toMatchSnapshot()
    })
    it('Testando mÃ©todo onChange utilizando uma mock', () => {
        const mockFn = jest.fn();
        const wrapper = shallow(<Picker value='' options={[]} />);
        wrapper.find('select').simulate('change',{target:{value:'bar'}});
        expect(mockFn).toBeCalled();
    })
})
