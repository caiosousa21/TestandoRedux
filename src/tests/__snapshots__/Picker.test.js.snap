// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index Testando interface com snapshots 1`] = `
<span>
  <h1>
    
  </h1>
  <select
    onChange={[Function]}
    value=""
  />
</span>
`;
